src/agent/communicator/src/http_socket.cpp

```diff
@@ -0,0 +1,254 @@
+#include "http_socket.hpp"
+
+#include "http_client_utils.hpp"
+#include <ihttp_socket.hpp>
+#include <logger.hpp>
+
+#include <boost/asio.hpp>
+#include <boost/asio/experimental/awaitable_operators.hpp>
+#include <boost/beast.hpp>
+#include <boost/system/error_code.hpp>
+
+#include <exception>
+#include <string>
+
+namespace http_client
+{
+    HttpSocket::HttpSocket(const boost::asio::any_io_executor& ioContext)
+        : m_socket(ioContext)
+    {
+    }
+
+    void HttpSocket::CallAsyncConnect(const boost::asio::ip::tcp::resolver::results_type& endpoints,
```

Comment:
Is this function necessary? It is only used once.

Reply:
It's meant to be overriden in unit tests. That's the only purpose of that function.

Reply:
It's gone now since the timeout was removed from synchronous Connect call